tests/test_component/test_task.py

```diff
@@ -117,18 +117,18 @@ def test_cpu_reqs(self):
         task.cpu_reqs.threads_per_process = threads_per_process
         self.assertEqual(task.cpu_reqs.cpu_threads, threads_per_process)
 
-        with self.assertRaises(TypeError):
+        with self.assertRaises(AttributeError):
```

Comment:
Hmm, shouldn't that remain a type error?  Looking at `https://github.com/radical-cybertools/radical.utils/blob/devel/src/radical/utils/typeddict.py#L346`, the error messsage does seem to indicate that a `TypeError` exception might be better suited?

Reply:
sigh.. you are right!! (yeah, that exception type is incorrect, made a corresponding RU PR [#339](https://github.com/radical-cybertools/radical.utils/pull/339))

Reply:
fixed (tests will pass after corresponding RU branch will be merged)